File: catkin_ws/src/turtlebot3_msgs/simple_navigation_goals/src/pub_initpose.py
#!/usr/bin/env python

import rospy
from geometry_msgs.msg import PoseWithCovarianceStamped

def talker():
	pub = rospy.Pub('/initialpose', PoseWithCovarianceStamped, queue_size=10)
	rospy.init_node('pub_initpose', anonymous=True)
	rate = rospy.Rate(10) # 10Hz

	pose = geometry_msgs.msg.PoseWithCovarianceStamped()
	pose.header.frame_id = "map"
	pose.pose.pose.position.x=5.0
	pose.pose.pose.position.y=5.0
	pose.pose.pose.position.z=0
	pose.pose.covariance=[0.25, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.25, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.06853891945200942]
	pose.pose.pose.orientation.w=1.0

	while not rospy.is_shutdown():
		rospy.loginfo(pose)
		pub.publish(initpose_msg)
    	rate.sleep()

File: catkin_ws/src/turtlebot3_msgs/simple_navigation_goals/src/pub_initpose.cpp
#include <ros/ros.h>
#include <geometry_msgs/PoseWithCovarianceStamped.h>

int main (int argc, char** argv) 
{
    ros::init(argc, argv, "pub_initpose");
    ros::NodeHandle nh_;
    ros::Publisher initPosePub_ = nh_.advertise<geometry_msgs::PoseWithCovarianceStamped>("initialpose", 2, true);
ros::Rate rate(1.0);
while(nh_.ok()) 
{
    //get time
    ros::Time scanTime_ = ros::Time::now();     
    //create msg
    geometry_msgs::PoseWithCovarianceStamped initPose_;
    //create the time & frame
    initPose_.header.stamp = scanTime_;
    initPose_.header.frame_id = "map";
    //position
    initPose_.pose.pose.position.x = 0.f;
    initPose_.pose.pose.position.y = 0.f;
    //angle
    initPose_.pose.pose.orientation.z = 0.f;
    //publish msg
    initPosePub_.publish(initPose_);
    //sleep
    rate.sleep();
}
return 0;
}
